etc/gen_gen/src/bin/gen_gen.rs

```diff
@@ -0,0 +1,324 @@
+// Copyright 2018 The Grin Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Main for building the genesis generation utility.
+
+use std::io::{BufRead, Write};
+use std::sync::Arc;
+use std::{fs, io, path, process};
+
+use chrono::prelude::Utc;
+use chrono::{Datelike, Duration, Timelike};
+use curl;
+use serde_json;
+
+use cuckoo_miner as cuckoo;
+use grin_chain as chain;
+use grin_core as core;
+use grin_miner_plugin as plugin;
+use grin_store as store;
+use grin_util as util;
+
+use grin_core::core::hash::Hashed;
+use grin_core::core::verifier_cache::LruVerifierCache;
+use grin_keychain::{BlindingFactor, ExtKeychain, Keychain};
+
+static BCHAIN_INFO_URL: &str = "https://blockchain.info/latestblock";
+static BCYPHER_URL: &str = "https://api.blockcypher.com/v1/btc/main";
+static BCHAIR_URL: &str = "https://api.blockchair.com/bitcoin/blocks?limit=2";
+
+static GENESIS_RS_PATH: &str = "../../core/src/genesis.rs";
+static PLUGIN_PATH: &str = "./cuckaroo_mean_cuda_29.cuckooplugin";
```

Comment:
In the README file you are saying

> Mine the block so we have at least a valid Cuckatoo Cycle.

Here you are using Cuckaroo to mine the block. Not so obvious what is going on here. Is the plan to mine the genesis block using the AR version so the next block that can be mined by everyone has a higher chance to be mined by the AF version? Is README wrong? What am I missing?